### PR TITLE
Add securityContext to stable/minecraft

### PR DESCRIPTION
**Pull request by 2vcps**
What this PR does / why we need it:
I screwed the PR up the first time. Trying again.

The docker image used in the chart stable/minecraft no longer runs its startup as root. When a volume is mapped for persistence the directory ends up being owned by root. Causing the startup script to fail with permission denied.

Which issue this PR fixes (optional, in fixes #(, fixes #<issue_number>, ...) format, will close that issue when PR gets merged): fixes #

Special notes for your reviewer:
I added security context to the values.yaml and the deployment.yaml template. This allows the pod to run as not root and also the fsGroup allows the user to write to the mounted directory at /data

### CODE REVIEW

File: stable/minecraft/values.yaml [hunk 1]
```diff
@@ -10,6 +10,10 @@ resources:
     memory: 512Mi
     cpu: 500m
 
+secContext:
```
Comment (unguiculus) [inline stable/minecraft/values.yaml]: Call it `securityContext`. No need to abbreviate here.

File: stable/minecraft/values.yaml [hunk 2]
```diff
@@ -10,6 +10,10 @@ resources:
...
+  # Securiyt context settings
```
Comment (unguiculus) [inline stable/minecraft/values.yaml]: Securiyt -> Security

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/minecraft/values.yaml (hunk 2 above)
Comment (2vcps) [reply stable/minecraft/values.yaml]: Thanks for fixes.

**Review by 2vcps — COMMENTED** [review]
Changes look good to me.

**Comment by 2vcps** [comment]
So I am new at all this, looks like the pull-charts-e2e is failing because the EULA is not being accepted.

helm install --name my-release \
    --set minecraftServer.eula=true stable/minecraft

How do we get the automated test to do this?

**Comment by 2vcps** [comment]
Not sure who else can move this along. 

[https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md](url)

Even points out Minecraft as a chart that provides the instructions in NOTES.txt for when it fails, but the chart is building correctly as it is outputting the need to accept the EULA.

**Comment by unguiculus** [comment]
You can add a custom values file for testing that accepts the EULA.

See https://github.com/kubernetes/charts/blob/master/test/README.md#providing-custom-test-values.

File: stable/minecraft/ci/test-values.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
+# ref: https://hub.docker.com/r/itzg/minecraft-server/
```
Comment (unguiculus) [inline stable/minecraft/ci/test-values.yaml]: Don't copy the complete `values.yaml`. Only add values which should override the defaults from `values.yaml`.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Not directly related to your PR, but the `Chart.yaml` file has `appVersion: 1.12.2` while `values.yaml` has `latest`. We want to have stable versions. So, while you are at it, could you please fix this?

**Review by unguiculus — APPROVED** [review]

File: stable/minecraft/ci/test-values.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# This must be overridden, since we can't accept this for the user.
+  eula: "TRUE"
```
Comment (unguiculus) [inline stable/minecraft/ci/test-values.yaml]: This must be `minecraftServer.eula`.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/minecraft/ci/test-values.yaml (hunk 4 above)
Comment (2vcps) [reply stable/minecraft/ci/test-values.yaml]: I made this change, sorry it took over a month.

**Comment by 2vcps** [comment]
Strange the circleCI is failing because it can't find this repo:
https://github.com/mikefarah/yaml/releases/download/1.13.1/yaml_linux_amd64
Not finding this anywhere in the helm chart, maybe it is a setting in CircleCI, that is above my pay grade.

**Comment by mattfarina** [comment]
@2vcps Two things:

1. There have been changes to the CI setup since this PR was created. Can you rebase from master to pick those up?
2. It appears that the distributor of the yaml tool we were using has removed the releases for the 1.x versions breaking the automated download of it. ick.

**Comment by 2vcps** [comment]
Ok, I think everything is fixed, I made the requested changes. What else can I do to get the "hold" removed?

**Pull request merged**